tests/gnmi/test_gnmi_smartswitch.py

```diff
@@ -0,0 +1,75 @@
+import json
+import logging
+import pytest
+import uuid
+from .helper import gnmi_set
+from dash_api.vnet_pb2 import Vnet
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+
+def get_vnet_proto(vni, guid):
+    pb = Vnet()
+    pb.vni = int(vni)
+    pb.guid.value = bytes.fromhex(uuid.UUID(guid).hex)
+    return pb.SerializeToString()
+
+
+def test_gnmi_appldb_01(duthosts, rand_one_dut_hostname, ptfhost):
+    '''
+    Verify GNMI native write with ApplDB
+    Update DASH_VNET_TABLE
+    '''
+    duthost = duthosts[rand_one_dut_hostname]
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    metadata = cfg_facts["DEVICE_METADATA"]["localhost"]
+    subtype = metadata.get('subtype', None)
+    type = metadata.get('type', None)
+    logger.info("type {}, subtype {}".format(type, subtype))
+    if type != "LeafRouter" or subtype != 'SmartSwitch':
+        pytest.skip("This test is supported only on smartswitch platforms")
+    # Locate the first online DPU
+    # DPU0            N/A              N/A         Online              up       N/A
+    target = None
+    result = duthost.shell("show chassis module status")
+    for line in result['stdout_lines']:
+        module_status = line.split()
+        if "Online" in module_status:
```

Comment:
>module_status [](http://example.com/codeflow?start=23&length=13)

This check is risky. Suggest
1. parse the table header line to get all the columns' names
2. get the index of expected column
3. loop all the remaining lines, check the expected column which have exact match of "Online"

Reply:
Updated.

Reply:
Please initialize name_idx, oper_status_idx to something (like NULL) before the loop, and check valid values after the loop.

I understand the python code is working, but not robust, and easy to trigger bug, if there is global variable with the same names.

Reply:
Fixed.